integration-tests/rewards_scripts_test.go

```diff
@@ -0,0 +1,394 @@
+	require.NoError(t, err)
+
+	tx, err = contractRewardsCoordinator.SubmitRoot(noSendTxOpts, root, 100)
+	require.NoError(t, err)
+
+	receipt, err = txMgr.Send(context.Background(), tx, true)
+	require.NoError(t, err)
+	require.Equal(t, receipt.Status, uint64(1))
+
+	// Process claim
+
+	indexToProve := 0
+
+	proof, err := generateEarnerMerkleProof(leaves, indexToProve)
+	require.NoError(t, err)
+
+	tokenProof, err := generateTokenMerkleProof(tokenLeaves, indexToProve)
+	require.NoError(t, err)
+
+	tokenIndices := make([]uint32, 1)
+	tokenProofs := make([][]byte, 1)
+	tokenProofs[0] = tokenProof
+
+	newTokenLeaves := make([]rewardsCoordinator.IRewardsCoordinatorTypesTokenTreeMerkleLeaf, 1)
+	newTokenLeaves[0] = defaultTokenLeaf(100, tokenAddr)
+
+	// this workflow assumes a new root submitted for every payment claimed.  So we get the latest rood index to process a claim for
+	rootLength, err := contractRewardsCoordinator.GetDistributionRootsLength(&bind.CallOpts{})
+	require.NoError(t, err)
+	rootIndex := rootLength.Uint64() - 1
+
+	claim := rewardsCoordinator.IRewardsCoordinatorTypesRewardsMerkleClaim{
+		RootIndex:       uint32(rootIndex),
+		EarnerIndex:     0,
+		EarnerTreeProof: proof,
+		EarnerLeaf:      earnerLeaves[indexToProve],
+		TokenIndices:    tokenIndices,
+		TokenTreeProofs: tokenProofs,
+		TokenLeaves:     newTokenLeaves,
+	}
+
+	testutils.AdvanceChainByNBlocks(1, anvilHttpEndpoint)
+
+	receipt, err = clients.ElChainWriter.ProcessClaim(context.Background(), claim, common.HexToAddress("0x01"), true)
+	require.NoError(t, err)
+	require.Equal(t, receipt.Status, uint64(1))
+
+	// After claim, claimer balance in strategy is zero
+	balanceAfterClaim, err := mockToken.BalanceOf(&bind.CallOpts{}, common.HexToAddress("0x0000000000000000000000000000000000000001"))
+	require.NoError(t, err)
+	assert.Equal(t, balanceAfterClaim.Int64(), amountPerPayment)
+}
+
+// These utils were inspired in those used in rewards scripts in Go Inc Squaring:
+// https://github.com/Layr-Labs/incredible-squaring-avs/blob/dev/contracts/script/utils/SetupDistributionsLib.sol
+func getEarners(deployer common.Address) []common.Address {
+	earners := make([]common.Address, 8)
+	for i := range earners {
+		earners[i] = deployer
+	}
+	return earners
+}
+
+func CreateEarnerLeaves(earners []common.Address, tokenLeaves [][32]byte) []rewardsCoordinator.IRewardsCoordinatorTypesEarnerTreeMerkleLeaf {
+	leaves := make([]rewardsCoordinator.IRewardsCoordinatorTypesEarnerTreeMerkleLeaf, len(earners))
+	tokenRoot := CreateTokenRoot(tokenLeaves)
+
+	for i, earner := range earners {
+		leaves[i] = rewardsCoordinator.IRewardsCoordinatorTypesEarnerTreeMerkleLeaf{
+			Earner:          earner,
+			EarnerTokenRoot: tokenRoot,
+		}
+	}
+	return leaves
+}
+
+func CreateTokenLeaves(
+	rewardsCoordinator *rewardsCoordinator.ContractRewardsCoordinator,
+	numTokenEarnings int,
+	tokenEarnings uint64,
```

Comment:
Let's rename these two
```suggestion
	numberOfTokenLeaves int,
	tokensEarned uint64,
```

Reply:
Done in [c84d2e6](https://github.com/Layr-Labs/eigensdk-go/pull/608/commits/c84d2e6fd56a24e33a6569e6bfc23c1ec13fe41b)